### PR TITLE
refactor(rag): switch to dify_config.

### PR DESCRIPTION
**Pull request by Poorandy**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [x] Please open an issue before creating a PR or link to an existing issue #5937 
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue. Close issue syntax: `Fixes #<issue number>`, see [documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) for more details.

Fixes 

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# Testing Instructions

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: api/core/rag/datasource/vdb/chroma/chroma_vector.py [hunk 1]
```diff
@@ -133,15 +133,14 @@ def init_vector(self, dataset: Dataset, attributes: list, embeddings: Embeddings
...
             config=ChromaConfig(
-                host=config.get('CHROMA_HOST'),
-                port=int(config.get('CHROMA_PORT')),
-                tenant=config.get('CHROMA_TENANT', chromadb.DEFAULT_TENANT),
-                database=config.get('CHROMA_DATABASE', chromadb.DEFAULT_DATABASE),
-                auth_provider=config.get('CHROMA_AUTH_PROVIDER'),
-                auth_credentials=config.get('CHROMA_AUTH_CREDENTIALS'),
+                host=dify_config.CHROMA_HOST,
+                port=dify_config.CHROMA_PORT,
+                tenant=dify_config.CHROMA_TENANT if config.get('CHROMA_TENANT') else chromadb.DEFAULT_TENANT,
```
Comment (laipz8200) [inline api/core/rag/datasource/vdb/chroma/chroma_vector.py]:
```suggestion
                tenant=dify_config.CHROMA_TENANT or chromadb.DEFAULT_TENANT,
```

File: api/core/rag/datasource/vdb/chroma/chroma_vector.py [hunk 2]
```diff
@@ -133,15 +133,14 @@ def init_vector(self, dataset: Dataset, attributes: list, embeddings: Embeddings
...
+                database=dify_config.CHROMA_DATABASE if config.get('CHROMA_DATABASE') else chromadb.DEFAULT_DATABASE,
```
Comment (laipz8200) [inline api/core/rag/datasource/vdb/chroma/chroma_vector.py]: ditto

File: api/core/rag/datasource/vdb/qdrant/qdrant_vector.py [hunk 3]
```diff
@@ -439,16 +439,15 @@ def init_vector(self, dataset: Dataset, attributes: list, embeddings: Embeddings
...
-        config = current_app.config
         return QdrantVector(
             collection_name=collection_name,
             group_id=dataset.id,
             config=QdrantConfig(
-                endpoint=config.get('QDRANT_URL'),
-                api_key=config.get('QDRANT_API_KEY'),
+                endpoint=dify_config.QDRANT_URL,
+                api_key=dify_config.QDRANT_API_KEY,
                 root_path=config.root_path,
```
Comment (laipz8200) [inline api/core/rag/datasource/vdb/qdrant/qdrant_vector.py:450]: Maybe need to put config back here.

**Review by laipz8200 — CHANGES_REQUESTED** [review]
Thanks! Please check my comments.

**Comment by Poorandy** [comment]
> Thanks! Please check my comments.谢谢！请查看我的评论。

Fixed.

**Review by laipz8200 — COMMENTED** [review]
Great!

**Review by laipz8200 — APPROVED** [review]

**Pull request merged**